src/irr/asset/CGLTFLoader.cpp

```diff
@@ -387,26 +411,46 @@ namespace irr
 
 					auto getShaders = [&](bool hasUV, bool hasColor) -> std::pair<core::smart_refctd_ptr<ICPUSpecializedShader>, core::smart_refctd_ptr<ICPUSpecializedShader>>
 					{
-						auto loadShaders = [&](const std::string_view& filename) -> core::smart_refctd_ptr<ICPUSpecializedShader>
+						auto loadShader = [&](const std::string_view& cacheKey) -> core::smart_refctd_ptr<ICPUSpecializedShader>
 						{
-							auto shader_bundle = assetManager->getAsset("include/irr/builtin/shaders/loaders/gltf/" + std::string(filename.data()), {});
-
-							bool status = !shader_bundle.isEmpty();
-							assert(status);
-
-							auto shader = core::smart_refctd_ptr_static_cast<ICPUSpecializedShader>(shader_bundle.getContents().begin()[0]);
-							return shader;
+							size_t storageSz = 1ull;
+							asset::SAssetBundle bundle;
+							const IAsset::E_TYPE types[]{ IAsset::ET_SPECIALIZED_SHADER, static_cast<IAsset::E_TYPE>(0u) };
+
+							assetManager->findAssets(storageSz, &bundle, cacheKey.data(), types);
+							if (bundle.isEmpty())
+								return nullptr;
+							auto assets = bundle.getContents();
+							
+							return core::smart_refctd_ptr_static_cast<ICPUSpecializedShader>(assets.begin()[0]);
 						};
 
 						if (hasUV) // if both UV and Color defined - we use the UV
-							return std::make_pair(loadShaders(SShaderPaths::UV::vertex), loadShaders(SShaderPaths::UV::fragment));
+							return std::make_pair(loadShader(VERT_SHADER_UV_CACHE_KEY), loadShader(FRAG_SHADER_UV_CACHE_KEY));
 						else if (hasColor)
-							return std::make_pair(loadShaders(SShaderPaths::Color::vertex), loadShaders(SShaderPaths::Color::fragment));
+							return std::make_pair(loadShader(VERT_SHADER_COLOR_CACHE_KEY), loadShader(FRAG_SHADER_COLOR_CACHE_KEY));
 						else
-							return std::make_pair(loadShaders(SShaderPaths::NoUVAndColor::vertex), loadShaders(SShaderPaths::NoUVAndColor::fragment));
+							return std::make_pair(loadShader(VERT_SHADER_NO_UV_COLOR_CACHE_KEY), loadShader(VERT_SHADER_NO_UV_COLOR_CACHE_KEY));
 					};
 
 					auto [cpuVertexShader, cpuFragmentShader] = getShaders(hasUV, hasColor);
+					size_t ds0_samplerBinding = 0, ds1_uboBinding = 0;
+
+					asset::ICPUDescriptorSetLayout::SBinding cpuSamplerBinding;
+
+					cpuSamplerBinding.binding = ds0_samplerBinding;
+					cpuSamplerBinding.type = EDT_COMBINED_IMAGE_SAMPLER;
+					cpuSamplerBinding.count = 1u;
+					cpuSamplerBinding.stageFlags = static_cast<ICPUSpecializedShader::E_SHADER_STAGE>(ICPUSpecializedShader::ESS_FRAGMENT);
+					cpuSamplerBinding.samplers = nullptr;
+
+					ICPUDescriptorSetLayout::SBinding cpuUboBinding;
+					cpuUboBinding.count = 1u;
+					cpuUboBinding.binding = ds1_uboBinding;
+					cpuUboBinding.stageFlags = static_cast<asset::ICPUSpecializedShader::E_SHADER_STAGE>(asset::ICPUSpecializedShader::ESS_VERTEX | asset::ICPUSpecializedShader::ESS_FRAGMENT);
+					cpuUboBinding.type = asset::EDT_UNIFORM_BUFFER;
```

Comment:
remember about the pipeline inputs metadata for this UBO (just how OBJ declares its inputs)